src/map.rs

```diff
@@ -423,6 +423,18 @@ where
         self.core.entry(hash, key)
     }
 
+    /// Get an entry in the map by index for in-place manipulation.
+    ///
+    /// Valid indices are *0 <= index < self.len()*
+    ///
+    /// Computes in **O(1)** time.
+    pub fn entry_at(&mut self, index: usize) -> Option<EntryAt<'_, K, V>> {
```

Comment:
The name is okay, but just to bikeshed a little, what do you think about `get_index_entry` and `IndexedEntry`? I think that may fit a little better with `get_index` and `get_index_mut`, and "get" methods in general returning `Option`, as well as similarity to the other adjective-`Entry` types.

Reply:
I dig `IndexedEntry`, and also renamed the method to `get_index_entry`. Some other options for the method would be `entry_index` (fits the existing `entry` method and `get`/`get_index` pattern, but feels odd) or `indexed_entry` (fits the return type). I don't have a strong opinion, but I'm happy to do another revision if somebody else has.